Helper/Data.php

```diff
@@ -1188,58 +1188,79 @@ public function getOpenInvoiceLineData(
      * @param integer|null $storeId
      * @return string the X API Key for the specified or current store
      */
-    public function getApiKey($storeId = null)
+    public function getPosApiKey($storeId = null)
     {
         if ($this->isDemoMode($storeId)) {
-            $apiKey = $this->_encryptor->decrypt(trim($this->getAdyenPosCloudConfigData('api_key_test')));
+            $apiKey = $this->_encryptor->decrypt(trim($this->getAdyenPosCloudConfigData('api_key_test', $storeId)));
         } else {
-            $apiKey = $this->_encryptor->decrypt(trim($this->getAdyenPosCloudConfigData('api_key_live')));
+            $apiKey = $this->_encryptor->decrypt(trim($this->getAdyenPosCloudConfigData('api_key_live', $storeId)));
         }
         return $apiKey;
     }
 
-    public function getPoiId()
+    /**
+     * Return the Terminal ID for the current store/mode
+     *
+     * @param int|null $storeId
+     * @return mixed
+     */
+    public function getPoiId($storeId = null)
     {
-        $poiId = $this->getAdyenPosCloudConfigData('pos_terminal_id');
+        $poiId = $this->getAdyenPosCloudConfigData('pos_terminal_id', $storeId);
         return $poiId;
     }
 
-    public function getAdyenMerchantAccount($method, $storeId)
+    /**
+     * Return the merchant account name configured for the proper payment method.
+     * If it is not configured for the specific payment method,
+     * return the merchant account name defined in required settings.
+     *
+     * @param $paymentMethod
+     * @param int|null $storeId
+     * @return string
+     */
+    public function getAdyenMerchantAccount($paymentMethod, $storeId = null)
     {
         $merchantAccount = $this->getAdyenAbstractConfigData("merchant_account", $storeId);
```

Comment:
as storeId can be null should we not retrieve it here or not make sure storeId needs to have an value ?

Reply:
I will revert it to mandatory, as in all the places where we call it, it is specified